### DISCUSSION TITLE
Is it possible to use form elements directly on blade views?

### DISCUSSION THREAD

**Discussion by bomshteyn** [filamentphp/filament#3447] [Help] [answered]
I have a complex create form for one of the models, which i created in custom livewire component, but i would love to be able to use the `forms::components.select` component for select with autocomplete instead of creating my own for consistency of look and feel.

I am trying to use the equivalent of this:
```PHP
Select::make('customer_id')
    ->label('Customer')
    ->required()
    ->options(Customer::all()->pluck('name', 'id'))
    ->searchable(),
```

I tried doing the following:
```PHP
    <forms::components.select
        name="customer"
        label="Customer"
        :options="$customers"
        required="true"
        searchable="true">
    </forms::components.select>
```

But it didn't work. 😞
Is there a way to achieve this?

**Comment by danharrin** [comment, marked as answer]
There isn't a way to use the components outside the form builder. Could you explain what makes your form too complex for the builder?

**Reply by bomshteyn** [reply]
This is how it looks:
![tbg-warehouse bc_pallets_build](https://user-images.githubusercontent.com/4259699/183761518-6f69b34f-7ce3-4b4e-b1ad-46f6e10591dd.png)

This the layout and functionality i am trying to achieve. 

- So it needs to have ability to add boxes, remove boxes to the pallet.
- add and remove items directly packaged onto the pallet
- sku field is an autocomplete field for products

Is this possible to achieve with the builder?

**Reply by danharrin** [reply]
Yup! Using the Repeater for boxes and items (boxes will use the grid() layout of the repeater), and then the sku could be a Select or a TextInput with datalist()

**Reply by danharrin** [reply]
The only thing you might struggle with is the "create" button position for each repeater, which would remain underneath the repeater. But that sounds like a small price to pay.

**Reply by bomshteyn** [reply]
Hmm, ok... Time to recreate in builder then... 

Is there no way to add a custom button to the builder to add an item to the repeater?

**Reply by danharrin** [reply]
You could add a `View::make('...')` component with a button view, and that button calls a custom Livewire method to add a new repeater item

**Reply by bomshteyn** [reply]
Thanks Dan!
